### DISCUSSION TITLE
Trying to install from source on a Raspberry Pi

### DISCUSSION THREAD

**Discussion by aaron-tan** [mitmproxy/mitmproxy#6065] [Q&A]
Hi everyone, I hope you're all doing well.

I'm currently trying to install mitmproxy from source on my Raspberry Pi but when running the command `venv/bin/pip install -e ".[dev]"` I get the following error: 
`ERROR: File "setup.py" not found. Directory cannot be installed in editable mode: /home/pi/mitmproxy
(A "pyproject.toml" file was found, but editable mode currently requires a setup.py based build.)
`

For reference, my python3 version is Python 3.9.2 and my pip version is pip 20.3.4

Thanks for any assistance.

**Comment by mhils** [comment]
1) Your setuptools version is outdated.
2) We've dropped support for Python 3.9 on main.

Hope this helps! :)
